### PR TITLE
Write `binary version -> commit hash` mapping to CI database (in private)

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 0b2c2741320ac542ccb3e744e8849ca4767e49ae with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61544/0b2c2741320ac542ccb3e744e8849ca4767e49ae/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/ci/clickhouse_helper.py [hunk 1]
```diff
@@ -109,6 +109,16 @@ def _insert_post(*args, **kwargs):
     def _insert_json_str_info(self, db, table, json_str):
         self.insert_json_str(self.url, self.auth, db, table, json_str)
 
+    def insert_json_into(self, db, table, json_str, safe=True):
```
Comment (Felixoid) [inline tests/ci/clickhouse_helper.py:112]: it's 100% the same as `ClickHouseHelper.insert_event_into`, but you manually do `json_str = json.dumps(data)`. Let's use `insert_event_into`.

File: tests/ci/ci.py [hunk 2]
```diff
@@ -1613,6 +1613,40 @@ def _upload_build_profile_data(
...
+
+    json_str = json.dumps(data)
+
+    print(f"::notice ::Log Adding record to versions history: {json_str}")
+
+    try:
+        ch_helper.insert_json_into(
+            db="default", table="version_history", json_str=json_str
+        )
+    except InsertException:
```
Comment (Felixoid) [inline tests/ci/ci.py:1646]: It will always be silent with default `safe=True`, so it's unnecessary try/except.

File: tests/ci/ci.py [hunk 3]
```diff
@@ -1986,6 +2020,16 @@ def main() -> int:
             ch_helper.insert_events_into(
                 db="default", table="checks", events=prepared_events
             )
+
+            if "DockerServerImage" in args.job_name and indata is not None:
+                _add_build_to_version_history(
+                    pr_info,
+                    job_report,
+                    indata["git_ref"],
```
Comment (Felixoid) [inline tests/ci/ci.py:2028]: `PRInfo.head_ref`? And one argument less

**Review by Felixoid — COMMENTED** [review]
Some points regarding making the changes less

**Review by Algunenano — APPROVED** [review]

**Comment by Felixoid** [comment]
`A Sync` is stuck. There's a merge-commit in the branch.

**Comment by nickitat** [comment]
> `A Sync` is stuck. There's a merge-commit in the branch.

and so what?

**Comment by Felixoid** [comment]
So you need to push there manually your changes. Either force-push the current head [eae047c](https://github.com/ClickHouse/ClickHouse/pull/61544/commits/eae047cd0b1182ecff316c1a5dc352ddb4738096), or merge it there

**Pull request merged**